test/e2e/dra/dra.go

```diff
@@ -775,6 +777,70 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 	})
 })
 
+// This test covers aspects of non graceful node shutdown by DRA controller
+// More details about this can be found in the KEP:
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2268-non-graceful-shutdown
+var _ = ginkgo.Describe("[sig-node] [Serial] [Disruptive] [Slow] [Feature:DynamicResourceAllocation] DRA: handling non graceful node shutdown", func() {
+	f := framework.NewDefaultFramework("dra")
+
+	// The driver containers have to run with sufficient privileges to
+	// modify /var/lib/kubelet/plugins.
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	ginkgo.Context("cluster", func() {
+		nodes := NewNodes(f, 2, 8)
+		driver := NewDriver(f, nodes, networkResources)
+		b := newBuilder(f, driver)
+
+		ginkgo.It("must deallocate on non graceful node shutdown", func(ctx context.Context) {
+			ginkgo.By("create test pod")
+			parameters := b.parameters()
+			label := "app.kubernetes.io/instance"
+			instance := f.UniqueName + "-test-app"
+			pod := b.podExternal()
+			pod.Labels[label] = instance
+			claim := b.externalClaim(resourcev1alpha2.AllocationModeWaitForFirstConsumer)
+			b.create(ctx, parameters, claim, pod)
+
+			ginkgo.By("wait for test pod " + pod.Name + " to run")
+			labelSelector := labels.SelectorFromSet(labels.Set(pod.Labels))
+			pods, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, pod.Namespace, labelSelector, 1, framework.PodStartTimeout)
+			framework.ExpectNoError(err, "start pod")
+			runningPod := &pods.Items[0]
+
+			nodeName := runningPod.Spec.NodeName
+			// Prevent builder tearDown to fail waiting for unprepared resources
+			delete(b.driver.Nodes, nodeName)
+			ginkgo.By("stop node " + nodeName + " non gracefully")
+			ginkgo.DeferCleanup(framework.RunCmd, "docker", "start", nodeName)
```

Comment:
There's a dependency on running on a kind cluster here which isn't expressed via some test tag. At least call it out in the test comment, as we don't have a feature tag defined for it.

Reply:
done